### PR TITLE
[BH-2033] Add fade in and fade out for relaxation

### PR DESCRIPTION
**Pull request by Maciej-Mudita**
Relaxation sounds will fade in and out
so that the songs play smoothly.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

File: module-audio/Audio/AudioCommon.hpp [hunk 1]
```diff
@@ -77,10 +77,17 @@ namespace audio
...
         Disable,
-        Enable
+        In,
+        InOut
+    };
+
+    struct FadeParams
+    {
+        Fade mode;
+        std::optional<std::chrono::seconds> playbackDuration = std::nullopt;
```
Comment (Lefucjusz) [inline module-audio/Audio/AudioCommon.hpp:90]: `std::optional` defaults to `std::nullopt`, there's no need to initialize it explicitly

File: products/BellHybrid/alarms/src/actions/PlayAudioActions.cpp [hunk 2]
```diff
@@ -23,13 +23,13 @@ namespace alarms
         if (duration.has_value()) {
             spawnTimer(duration.value());
         }
-        const auto fadeInSettings = utils::getNumericValue<bool>(
+        const auto fadeSettings = utils::getNumericValue<bool>(
```
Comment (Lefucjusz) [inline products/BellHybrid/alarms/src/actions/PlayAudioActions.cpp:26]:
```suggestion
        const auto fadeEnabledInSettings = utils::getNumericValue<bool>(
```
That way it's more self-explanatory, `fadeSettings` might suggest that this is an object containing multiple settings, not just a simple boolean flag.

File: products/BellHybrid/services/audio/VolumeFade.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+#include "VolumeFade.hpp"
+#include <log/log.hpp>
+
+namespace audio
+{
+    namespace
+    {
+        constexpr auto timerName{"volumeFadeTimer"};
+        constexpr std::chrono::milliseconds fadeInterval{33};
```
Comment (Lefucjusz) [inline products/BellHybrid/services/audio/VolumeFade.cpp:12]:
```suggestion
        using namespace std::chrono_literals;
        constexpr auto fadeInterval{33ms};
```
It's easier to read and should be functionally equivalent :wink:

File: products/BellHybrid/services/audio/VolumeFade.cpp [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            using floatToMs         = std::chrono::duration<float, std::chrono::milliseconds::period>;
+            const auto timeRequired = floatToMs(currentVolume / fadeStep) * fadeInterval.count();
+            const auto calculatedTime =
+                std::chrono::duration_cast<std::chrono::milliseconds>(duration - elapsed - timeRequired);
+            return calculatedTime > std::chrono::milliseconds::zero() ? calculatedTime
+                                                                      : std::chrono::milliseconds::zero();
+        }
+    } // namespace
+
+    VolumeFade::VolumeFade(sys::Service *parent, SetCallback callback) : setVolumeCallback(callback)
```
Comment (Lefucjusz) [inline products/BellHybrid/services/audio/VolumeFade.cpp:28]:
```suggestion
    VolumeFade::VolumeFade(sys::Service *parent, SetCallback callback) : setVolumeCallback(std::move(callback))
```

**Review by Lefucjusz — APPROVED** [review]

File: module-audio/Audio/AudioCommon.hpp (hunk 1 above)
Comment (Maciej-Mudita) [reply module-audio/Audio/AudioCommon.hpp:90]: done

File: products/BellHybrid/services/audio/VolumeFade.cpp (hunk 3 above)
Comment (Maciej-Mudita) [reply products/BellHybrid/services/audio/VolumeFade.cpp:12]: done

File: products/BellHybrid/services/audio/VolumeFade.cpp (hunk 4 above)
Comment (Maciej-Mudita) [reply products/BellHybrid/services/audio/VolumeFade.cpp:28]: done

**Review by dawidwojtas-mudita — APPROVED** [review]

**Pull request merged**